File: downloads-that-dont-fail-automating-file-verification/python/pdf_verification.py
import PyPDF2
import os

# Define the expected file name and path
expected_file_name = "sample-file.pdf"
download_directory = "/path/to/downloads"
file_path = os.path.join(download_directory, expected_file_name)

# Open the downloaded PDF file
with open(file_path, 'rb') as file:
    pdf_reader = PyPDF2.PdfReader(file)
    first_page = pdf_reader.pages[0]
    text = first_page.extract_text()

# Check if the PDF contains expected content
if "expected content" in text:
    print("PDF content verification passed")
else:
    raise Exception("PDF content verification failed")

File: downloads-that-dont-fail-automating-file-verification/javascript/pdfVerification.js
const path = require("path");
const fs = require("fs");
const pdf = require("pdf-parse");

// Define the expected file name and path
const expectedFileName = "sample-file.pdf";
const downloadDirectory = "/path/to/downloads/";
const filePath = path.join(downloadDirectory, expectedFileName);

async function verifyPdfContent(filePath) {
  // Read the PDF file
  const dataBuffer = fs.readFileSync(filePath);

  // Parse the PDF content
  const data = await pdf(dataBuffer);

  // Extract text from the first page
  const text = data.text;

  // Check if the PDF contains the expected content
  if (text.includes("expected content")) {
    console.log("PDF content verification passed");
  } else {
    throw new Error("PDF content verification failed");
  }
}

verifyPdfContent(filePath);
